test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Data;
+using System.Data.SqlClient;
+using Microsoft.ML.Data;
+using Microsoft.ML.TestFramework;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class DatabaseLoaderTests : BaseTestClass
+    {
+        public DatabaseLoaderTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void TryItOut()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var connectionString = @"Server=(localdb)\mssqllocaldb;Database=EFGetStarted.AspNetCore.NewDb;Trusted_Connection=True;ConnectRetryCount=0";
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+
+                using (SqlCommand command = new SqlCommand(
+                    "SELECT SepalLength, SepalWidth, PetalLength, PetalWidth, Label FROM IrisData",
+                    connection))
+                {
+                    DatabaseLoader loader = new DatabaseLoader(mlContext, new DatabaseLoader.Options()
+                    {
+                        Columns = new[]
+                        {
+                            new DatabaseLoader.Column() { Name = "SepalLength", Type = DbType.Single },
+                            new DatabaseLoader.Column() { Name = "SepalWidth", Type = DbType.Single },
+                            new DatabaseLoader.Column() { Name = "PetalLength", Type = DbType.Single },
+                            new DatabaseLoader.Column() { Name = "PetalWidth", Type = DbType.Single },
+                            new DatabaseLoader.Column() { Name = "Label", Type = DbType.Int32 },
+                        }
+                    });
+
+                    IDataView trainingData = loader.Load(() => command.ExecuteReader());
```

Comment:
Once we have this foundational API we also need to add it to the MLContext catalog, so having convenient methods such as:

`mlContext.Data.LoadFromDbDataReader<ModelInputData>(() => command.ExecuteReader());`

or

```
IDataView trainingDataView = mlContext.Data.LoadFromDbTable<ModelInputData, SqlConnection>(connString: myConnString, 
                                                                                                 tableName: "TrainingDataTable"); 
```